src/olympia/search/filters.py

```diff
@@ -471,12 +472,38 @@ def filter_queryset(self, request, qs, view):
         order_by = None
 
         if sort_param is not None:
+            split_sort_params = sort_param.split(',')
             try:
                 order_by = [self.SORTING_PARAMS[name] for name in
-                            sort_param.split(',')]
+                            split_sort_params]
             except KeyError:
                 raise serializers.ValidationError('Invalid "sort" parameter.')
 
+            # Random sort is a bit special.
+            # First, it can't be combined with other sorts.
+            if 'random' in split_sort_params and len(split_sort_params) > 1:
+                raise serializers.ValidationError(
+                    'The "random" "sort" parameter can not be combined.')
+
+            # Second, for perf reasons it's only available when the 'featured'
+            # param is present (to limit the number of documents we'll have to
+            # apply the random score to) and a search query is absent
+            # (to prevent clashing with the score functions coming from a
+            # search query).
+            if sort_param == 'random':
+                is_random_sort_available = (
+                    AddonFeaturedQueryParam.query_param in request.GET and
+                    not search_query_param
+                )
+                if is_random_sort_available:
+                    qs = qs.query(
+                        'function_score', functions=[query.SF('random_score')])
+                else:
+                    raise serializers.ValidationError(
+                        'The "random" "sort" parameter can only be specified '
```

Comment:
I'd write that a bit different: `The "sort" parameter "random" can only be specified...` but that's just personal preference.